doc/deployment_guide.md

```diff
@@ -0,0 +1,797 @@
+
+### MDIP Operations
+
+A system is said to be an MDIP Client if it implements MDIP Keymaster functionality. The MDIP Keymaster library and tools made available in this repository implement functionality to create valid MDIP operations. MDIP Keymaster is available as a REST OpenAPI interface, Command Line Interface, Web User Interface, NodeJS and Python SDKs, npm library, with more additions planned.
+
+Each Keymaster Client manages a **private seed and key** used to sign or decrypt MDIP operations for a specific wallet.
+
+MDIP nodes that need to issue MDIP Challenges, verify Responses, issue Credentials, etc will need an MDIP Keymaster to manage the server's MDIP Wallet containing the keys to the server's identity on the network.
+
+MDIP operators can interact with their server's Keymaster in multiple ways:
+
+- use the `./kc` CLI tool or
+- use the Keymaster WebUI at `http://localhost:4226` or
+- use the Keymaster REST OpenAPI at `http://localhost:4226/api/v1/`
+
+The MDIP [Integration Guide](integration_guide.md) focuses on Keymaster deployments, Keymaster OpenAPI interface, and integration of MDIP wallet functionality with 3rd party applications.
+
+### MDIP Components
+
+![MDIP-components](https://github.com/user-attachments/assets/c88a33f0-aa61-48b9-a62a-ca956f1038d9)
+
+All the MDIP components illustrated above are composed together using Docker Compose in the official ['kc' v.0.3-beta repository](https://github.com/KeychainMDIP/kc/releases/tag/v0.3-beta). The repository provides a simple [`start-node`](https://github.com/KeychainMDIP/kc/blob/v0.3-beta/start-node) script that will compile, dockerize, and launch all required MDIP components. The convenience of this script will be preserved as we explore configuration and deployment options below.
+
+Note that each MDIP components can also be manually launched using native nodeJS commands. The Dockerfile for each component will reveal the particular shell command being executed within the Docker environment. Using the [Dockerfile.gatekeeper](https://github.com/KeychainMDIP/kc/blob/v0.3-beta/Dockerfile.gatekeeper) as an example, we see the command used to launch the Gatekeeper API:
+`CMD ["node", "server/src/gatekeeper-api.js"]`
+
+While perhaps not optimal for a production environment, developement environments can be deployed with all the required components on a single moderately-sized server. The statistics below demonstrate a typical development environment:
+
+![MDIP-resources](https://github.com/user-attachments/assets/7d63d4dc-fa51-4254-b9ad-94b91d98f78d)
+
+- kc-gatekeeper-1 is the [MDIP Gatekeeper](https://keychain.org/docs/mdip/scheme) API process.
+- kc-keymaster-1 is a [Keymaster API](https://keychain.org/docs/keychain/library/api/) to manage the server's MDIP identity.
```

Comment:
Similarly, the keymaster service is more than an API; it manages the wallet, so the both the description and reference to the API doc should be adjusted.